### DISCUSSION TITLE
x-forwarded-for: define number of trusted proxies?

### DISCUSSION THREAD

**Discussion by thomas-riccardi** [keycloak/keycloak#12356] [Archive]
When deploying keycloak behind a http load balancer, we set `proxy=edge` so keyloak accesses the client IP from `x-forwarded-for` header.

However, this opens a security issue as the value cannot be fully trusted: reverse proxies *append* values to the header.
The usual way to go is to configure a trusted proxy count (or list), see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#selecting_an_ip_address

How can we define the trusted proxies count with the quarkus distribution of keycloak?

https://www.keycloak.org/server/reverseproxy doesn't talk about it, nor a search in github code/issues/discussions or the user mailing list.

**Comment by markszabo** [comment]
We are facing the same issue. We are running Keycloak behind an AWS ALB, but since [AWS ALBs can only preserve, append or remove](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/x-forwarded-headers.html#x-forwarded-for) (but not replace) the `X-Forwarded-For` header, in the end if a user sets the `X-Forwarded-For` header, then Keycloak will pick up the IP the user specified and use it for IP filtering and also for logging. This is a serious security problem for us. 

Having the trusted proxy count option would solve this issue.
